packages/loader/container-loader/src/container.ts

```diff
@@ -815,17 +817,16 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
                 this._attachState = AttachState.Attaching;
                 this.emit("attaching");
             }
-            assert(!!this.cachedAttachSummary,
+            assert(!!this.cachedAttachSummary || this.loader.services.detachedBlobStorage?.size !== 0,
```

Comment:
I worry that number of states and number of failure points increases with new design where it would be hard to maintain same kind of retry logic RE caching this.cachedAttachSummary. I think for all retryable errors we already have solution (usage of runWithRetry), and non-retryable errors should not be retried, and thus this.cachedAttachSummary removed (as a way to re-enter attach() process).

@jatgarg - what do you think?

Also see my other comments RE reverting who drives the process - I think if runtime does it, it has more control of making it retryable if it wants to.

Reply:
Yes since we retry creating container we can avoid caching this summary as the host does not need to retry anymore.